impl/src/main/java/org/jboss/weld/bootstrap/events/ContainerLifecycleEventPreloader.java

```diff
@@ -51,12 +51,15 @@ public Void call() throws Exception {
         }
     }
 
+    // Keep static to only initialize the group once.
+    private static final ThreadGroup THREAD_GROUP = new ThreadGroup("weld-preloaders");
+
     private final ExecutorService executor;
     private final ObserverNotifier notifier;
 
     public ContainerLifecycleEventPreloader(int threadPoolSize, ObserverNotifier notifier) {
         this.executor = Executors.newFixedThreadPool(threadPoolSize,
-                new DaemonThreadFactory(new ThreadGroup("weld-preloaders"), "weld-preloader-"));
+                new DaemonThreadFactory(THREAD_GROUP, "weld-preloader-"));
```

Comment:
I think that in this case we should just avoid using the `ThreadGroup` completely. This `ExecutorService` is a short-lived construct and we don't need to group the threads at all.